controllers/component_dependency_update_controller.go

```diff
@@ -488,6 +539,145 @@ func (r *ComponentDependencyUpdateReconciler) removePipelineFinalizer(ctx contex
+			} else {
+				if repoAuth.Auth == "" {
+					log.Error(fmt.Errorf("password and username and auth are empty in auth config for repository %s", repoName), "no valid auth")
+					continue
+				} else {
+					decodedAuth, err := base64.StdEncoding.DecodeString(repoAuth.Auth)
+					if err != nil {
+						log.Error(err, fmt.Sprintf("unable to decode docker config json auth for repository %s in the secret %s", repoName, secret.Name))
+						continue
+					}
+					authParts := strings.Split(string(decodedAuth), ":")
+					filteredSecretsData = append(filteredSecretsData, RepositoryCredentials{SecretName: secret.Name, RepoName: repoName, UserName: authParts[0], Password: authParts[1]})
+				}
+			}
+		}
+	}
+
+	imageRepositoryUsername, imageRepositoryPassword, err := r.GetCredentialsForImageRepository(ctx, updatedOutputImage, filteredSecretsData)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("unable to find credential for repository %s", updatedOutputImage))
+		return "", "", err
+	}
+	return imageRepositoryUsername, imageRepositoryPassword, nil
+}
+
+// GetCredentialsForImageRepository This method returns credentials for image repository
+func (r ComponentDependencyUpdateReconciler) GetCredentialsForImageRepository(ctx context.Context, outputImage string, imageRepoSecrets []RepositoryCredentials) (string, string, error) {
+	log := ctrllog.FromContext(ctx)
+
+	repoPath := outputImage
+	if strings.Contains(outputImage, "@") {
+		repoPath = strings.Split(outputImage, "@")[0]
+	} else {
+		if strings.Contains(outputImage, ":") {
+			repoPath = strings.Split(outputImage, ":")[0]
+		}
+	}
+	repoPath = strings.TrimSuffix(repoPath, "/")
+
+	username := ""
+	password := ""
+	// first check for credential which matches full repository path
+	for _, credential := range imageRepoSecrets {
+		credentialRepo := strings.TrimSuffix(credential.RepoName, "/")
+		if repoPath == credentialRepo {
+			log.Info("found full match of repository in auth", "repo", repoPath, "secretName", credential.SecretName)
+			username = credential.UserName
+			password = credential.Password
+			break
+		}
+	}
+	if username != "" && password != "" {
+		return username, password, nil
+
+	}
+
+	// check for partial match, get the most complete match
+	// if there is multiple secrets for registry and some partial match, upload sbom would fail anyway, because it chooses them randomly (until cosign fixes it)
+	repoParts := strings.Split(repoPath, "/")
+	for {
+		repoParts = repoParts[:len(repoParts)-1]
+		if len(repoParts) == 0 {
+			break
+		}
+		partialRepo := strings.Join(repoParts, "/")
+
+		for _, credential := range imageRepoSecrets {
+			credentialRepo := strings.TrimSuffix(credential.RepoName, "/")
+			if partialRepo == credentialRepo {
+				log.Info("partial match found of repository in auth", "repo", partialRepo, "secretName", credential.SecretName)
+				username = credential.UserName
+				password = credential.Password
```

Comment:
Do we need to check them for emptiness?

Reply:
at this point we didn't find full match

and checking for partial

if there is any auth which has base64 auth without user or password builds would be failing anyway

Reply:
adding